filebeat/beat/spooler.go

```diff
@@ -14,18 +15,18 @@ type Spooler struct {
 	nextFlushTime time.Time
 	spool         []*input.FileEvent
 	Channel       chan *input.FileEvent
+	mutex         *sync.Mutex
```

Comment:
To make it clear what the mutex guards, consider naming it `nextFlushTimeMutex` (long but clear) and declare it next to `nextFlushTime`. It helps so that someone reading the code will know that the mutex must be used to read/write from `nextFlushTime`.